cmd/juju/resource/deploy.go

```diff
@@ -236,7 +236,7 @@ func (d deployUploader) uploadFile(resourcename, filename string) (id string, er
 }
 
 func (d deployUploader) uploadDockerDetails(resourcename, registryPath string) (id string, error error) {
-	data, err := yaml.Marshal(resources.DockerImageDetails{
+	data, err := json.Marshal(resources.DockerImageDetails{
```

Comment:
Add a TODO to allow a  username and password to be provided.
Or actually, we should really check that the --resource arg is not a filename right? If it's a filename, we can just read the content of that as the resource content as would be obtained from the charm store, including username, password.